src/test/java/bio/terra/workspace/integration/common/auth/AuthService.java

```diff
@@ -0,0 +1,79 @@
+package bio.terra.workspace.integration.common.auth;
+
+import bio.terra.workspace.integration.common.configuration.TestConfiguration;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.*;
+import java.util.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthService {
+
+  private final TestConfiguration testConfig;
+  private final Map<String, String> userTokens = new HashMap<>();
+  private final List<String> domainWideDelegationAccessScopes =
+      Arrays.asList(
+          "openid",
+          "email",
+          "profile",
+          "https://www.googleapis.com/auth/devstorage.full_control",
+          "https://www.googleapis.com/auth/cloud-platform");
+
+  @Autowired
+  public AuthService(TestConfiguration testConfig) {
+    this.testConfig = testConfig;
+  }
+
+  public String getAuthToken(String userEmail) throws IOException, InterruptedException {
+    if (!userTokens.containsKey(userEmail)) {
+      String vaultPath = testConfig.getVaultPath();
+      userTokens.put(userEmail, getDomainWideDelegationAccessToken(userEmail, vaultPath));
+    }
+    return userTokens.get(userEmail);
+  }
+
+  private String getDomainWideDelegationAccessToken(
+      String userEmail, String serviceAccountVaultPath) throws IOException, InterruptedException {
+    String serviceAccountJson = getJsonSecretFromVault(serviceAccountVaultPath);
+    GoogleCredentials credentials =
+        GoogleCredentials.fromStream(new ByteArrayInputStream(serviceAccountJson.getBytes()))
+            .createScoped(domainWideDelegationAccessScopes)
+            .createDelegated(userEmail);
+    credentials.refreshIfExpired();
+    AccessToken newAccessToken = credentials.getAccessToken();
+    return newAccessToken.getTokenValue();
+  }
```

Comment:
I don't think the test should read from Vault at runtime. This should use an architecture similar to our apps, where users (including Jenkins) are required to render configs first before executing the code, and all secrets are written to config files. This would also eliminate the need for a vault token inside runtime code. So, to run tests, we'd need a flow like (high level):
1. User generates configs - application.properties, service-account.json, etc - by injecting Vault secrets into templatized config files. This only needs to be done once per environment, or when configs change.
2. Run tests via `gradlew integrationTest` or similar, which read directly from the rendered config files and therefore don't need Vault access

Reply:
This should be doable for this PR. I may reach out to you for further discussion on it.